app/common/ConfigAPI.ts

```diff
@@ -0,0 +1,31 @@
+import {BaseAPI, IOptions} from "app/common/BaseAPI";
+import {addCurrentOrgToPath} from 'app/common/urlUtils';
+
+/**
+ * An API for accessing the internal Grist configuration, stored in
+ * config.json.
+ */
+export class ConfigAPI extends BaseAPI {
+  constructor(private _homeUrl: string, options: IOptions = {}) {
+    super(options);
+  }
+
+  public async getValue(key: string): Promise<any> {
+    return (await this.requestJson(`${this._url}/api/config/${key}`, {method: 'GET'})).value;
+  }
+
+  public async setValue(value: any, restart=false): Promise<void> {
+    await this.request(`${this._url}/api/config`, {
+      method: 'PATCH',
+      body: JSON.stringify({config: value, restart}),
+    });
+  }
+
+  public async restartServer(): Promise<void> {
+    await this.request(`${this._url}/api/admin/restart`, {method: 'POST'});
```

Comment:
The restart backend will return a 409 if it can't contact the parent process. This might be enough in some situations, however, `yarn start` in local development will start the process with nodemon, which is a parent process.

I spent way too long seeing if nodemon could also be sent a message to restart, but I have concluded: not in the way we're using it. It would only work if we had another wrapper for nodemon. I don't think it's worthwhile, as this really only affects local development. The Docker images do restart correctly.

Reply:
As to your question of throwing an error message: I can't see a way to distinguish `supervisor.mjs` from `nodemon` as parent processes. I could get the parent process ID and then try to get the name by executing `ps`, and using that to throw a better error message.

Reply:
Could you send a `grist:ping` message to parent, and wait for a `grist:pong` message back? Is that too much?

Reply:
Or have the supervisor set a GRIST_IS_RUNNING_UNDER_SUPERVISOR environment variable before running Grist?

Reply:
Oh. That's easy, why didn't I think of that. How's that? https://github.com/gristlabs/grist-core/pull/1095/commits/fd866e6d8273f3e6780ac76dfa2de858c6738074